zenoh/src/api/info.rs

```diff
@@ -104,6 +120,347 @@ impl SessionInfo {
+            | LinkAuthId::Ws => None, // avoid using _ wildcard to ensure that new protocols are correctly handled
+        };
+        let priorities = link
+            .priorities
+            .as_deref()
+            .map(|p| (*p.start() as u8, *p.end() as u8))
+            .unwrap_or((
+                // max priority is the lowest numerical value
+                zenoh_protocol::core::Priority::MAX as u8,
+                zenoh_protocol::core::Priority::MIN as u8,
+            ));
+        // Link reliability is always known - it's either specified explicitly in
+        // the config or inferred from the protocol
+        let reliability = link.reliability.unwrap_or_else(|| {
+            let inspector = zenoh_link::LocatorInspector::default();
+            // TODO: do we need to check both?
+            if inspector.is_reliable(&link.src).unwrap_or(false)
+                && inspector.is_reliable(&link.dst).unwrap_or(false)
+            {
+                Reliability::Reliable
+            } else {
+                Reliability::BestEffort
+            }
+        });
+        Link {
+            zid,
+            src: link.src.clone(),
+            dst: link.dst.clone(),
+            group: link.group.clone(),
+            mtu: link.mtu,
+            is_streamed: link.is_streamed,
+            interfaces: link.interfaces.clone(),
+            auth_identifier,
+            priorities,
+            reliability,
+        }
+    }
+
+    /// Gets the ZenohId of the transport this link belongs to.
+    #[inline]
+    pub fn zid(&self) -> &ZenohId {
+        &self.zid
+    }
+
+    /// Gets the source locator (local endpoint).
+    #[inline]
+    pub fn src(&self) -> &Locator {
+        &self.src
+    }
+
+    /// Gets the destination locator (remote endpoint).
+    #[inline]
+    pub fn dst(&self) -> &Locator {
+        &self.dst
+    }
+
+    /// Gets the optional group locator (destination if link is for multicast).
+    #[inline]
+    pub fn group(&self) -> Option<&Locator> {
+        self.group.as_ref()
+    }
+
+    /// Gets the maximum transmission unit in bytes (MTU) of the link.
+    #[inline]
+    pub fn mtu(&self) -> u16 {
+        self.mtu
+    }
+
+    /// Returns whether the link is streamed.
+    #[inline]
+    pub fn is_streamed(&self) -> bool {
+        self.is_streamed
+    }
+
+    /// Gets the network interfaces associated with the link.
+    #[inline]
+    pub fn interfaces(&self) -> &[String] {
+        &self.interfaces
+    }
+
+    /// Gets the authentication identifier used for the link (if any).
+    #[inline]
+    pub fn auth_identifier(&self) -> Option<&str> {
+        self.auth_identifier.as_deref()
+    }
+
+    /// Gets the optional priority range (min, max) associated with the link.
+    /// The numeric priority values corresponds [`Priority`](crate::qos::Priority)
+    /// but can also contain value 0 (Control) not exposed in the public enum.
+    #[inline]
+    pub fn priorities(&self) -> (u8, u8) {
```

Comment:
Maybe `priorities()` and `reliability()` should return `None` if the associated transport has `is_qos() == false`.

Reply:
Done